routers/repo/repo.go

```diff
@@ -311,7 +311,7 @@ func Download(ctx *context.Context) {
 			ctx.Handle(500, "GetTagCommit", err)
 			return
 		}
-	} else if len(refName) == 40 {
+	} else if len(refName) > 7 && len(refName) <= 40 {
```

Comment:
It should be == 7 or == 40, beside that I'm not sure if this is the only place that needs to be changed